src/primitives.ts

```diff
@@ -220,6 +220,44 @@ function split<T extends string, D extends string = ''>(str: T, delimiter?: D) {
   return str.split(delimiter ?? ('' as const)) as Split<T, D>
 }
 
+/**
+ * Checks if a string starts with another string.
+ * T: The string to check.
+ * S: The string to check against.
+ * P: The position to start the search.
+ */
+type StartsWith<
+  T extends string,
+  S extends string,
+  P extends number = 0,
+> = Math.IsPositive<P> extends true
+  ? P extends 0
+    ? T extends `${infer TFirst}${infer TRest}`
+      ? S extends `${infer SFirst}${infer SRest}`
+        ? TFirst extends SFirst
+          ? StartsWith<TRest, SRest, P> // Compare next character
+          : false // T differs from S
+        : true // S (Search) is exhausted (and didn't fail)
+      : false // T (Text) is exhausted with S (Search) leftover
+    : StartsWith<Slice<T, P>, S, 0> // P is >0, slice
+  : StartsWith<T, S, 0> // P (Position) is negative, ignore
+
```

Comment:
Wishing for a more succinct way to express this type, but `position` creates a bunch of extra if/else blocks and corner cases

Reply:
Is positioning part of the native API? I didn't know about it.
In this case I think you can use Slice in to simplify the type implementation

Reply:
It is, but the way `startsWith` and `slice` handle negative indices is a little different

Reply:
<img width="246" alt="Screenshot 2023-10-04 at 6 01 50 PM" src="https://github.com/gustavoguichard/string-ts/assets/33426811/281f1296-5127-4119-a0a5-86d0b3fdc148">

Reply:
I tried to capture all these cases in the unit tests